hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/third_party/uriparser2/uriparser2/uriparser2.c

```diff
@@ -71,9 +71,11 @@ static const char *copy_path(const UriPathSegmentA *ps, char **buffer) {
 static int parse_int(const char *first, const char *after_last) {
 	const int size = after_last - first;
 	if (size) {
-		char buffer[size + 1];
+		char* buffer = (char*) malloc(size + 1);
```

Comment:
Isn't the compiler being too pedantic? The old code should be safe.

Reply:
Local arrays are stored on the stack. The stack has a fixed limit. Thus, MSVC tries to play safe by making the size a compile-time requirement, forcing us to use the heap. I don't know how GCC solves this though.